### PR TITLE
Add soversion to msquic

### PR DESCRIPTION
**Pull request by thhous-msft**

### CODE REVIEW

**Review by nibanks — DISMISSED** [review]

File: scripts/make-packages.sh [hunk 1]
```diff
@@ -94,37 +94,42 @@ fi
...
 
   # Debian/Ubuntu
   if [ "$LIBDIR" == 'lib64' ]; then
       LIBDIR="lib/x86_64-linux-gnu"
   fi
-  FILES="${ARTIFACTS}/libmsquic.${LIBEXT}=/usr/${LIBDIR}/libmsquic.${LIBEXT}"
-  if [ -e "$ARTIFACTS/libmsquic.lttng.${LIBEXT}" ]; then
-     FILES="${FILES} ${ARTIFACTS}/libmsquic.lttng.${LIBEXT}=/usr/${LIBDIR}/libmsquic.lttng.${LIBEXT}"
+  FILES="${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}=/usr/${LIBDIR}/libmsquic.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}"
+  FILES="${FILES} ${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}=/usr/${LIBDIR}/libmsquic.lttng.${LIBEXT}.${VER_MAJOR}"
```
Comment (wfurt) [inline scripts/make-packages.sh:118]:
```suggestion
  FILES="${FILES} ${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}=/usr/${LIBDIR}/libmsquic.${LIBEXT}.${VER_MAJOR}"
```

File: scripts/make-packages.sh [hunk 2]
```diff
@@ -94,37 +94,42 @@ fi
...
 if [ "$OS" == "linux" ]; then
+  # Create symlink
+  ln -s "${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}" "${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}"
+
   # RedHat/CentOS
-  FILES="${ARTIFACTS}/libmsquic.${LIBEXT}=/usr/${LIBDIR}/libmsquic.${LIBEXT}"
-  if [ -e "$ARTIFACTS/libmsquic.lttng.${LIBEXT}" ]; then
-     FILES="${FILES} ${ARTIFACTS}/libmsquic.lttng.${LIBEXT}=/usr/${LIBDIR}/libmsquic.lttng.${LIBEXT}"
+  FILES="${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}=/usr/${LIBDIR}/libmsquic.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}"
+  FILES="${FILES} ${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}=/usr/${LIBDIR}/libmsquic.lttng.${LIBEXT}.${VER_MAJOR}"
```
Comment (wfurt) [inline scripts/make-packages.sh:102]:
```suggestion
  FILES="${FILES} ${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}=/usr/${LIBDIR}/libmsquic.${LIBEXT}.${VER_MAJOR}"
```

**Review by wfurt — APPROVED** [review]
Looks good with proposed changes. (looks like typo) 

Work fine with https://github.com/dotnet/runtime/pull/61021

**Comment by omajid** [comment]
> The issue I'm running into now is it doesn't look like fpm actually supports doing the ldconfig as a postinstall script. 

Reading the docs, `--rpm-posttrans FILE` looks like a way to run `FILE` as a script during posttrans phase. Perhaps putting a call to ldconfig inside a script and then using it with the `--rpm-posttrans` option will work?

File: scripts/package-distribution.ps1 [hunk 3]
```diff
@@ -19,6 +19,8 @@ $ArtifactsBinDir = Join-Path $BaseArtifactsDir "bin"
 # All direct subfolders are OS's
 $Platforms = Get-ChildItem -Path $ArtifactsBinDir
 
+$Version = "1.10.0"
```
Comment (anrossi) [inline scripts/package-distribution.ps1:22]: Is there a way we can keep this in sync with the version in the source code? Or a script to update both together?

Comment (thhous-msft) [reply scripts/package-distribution.ps1:22]: The update-version script was updated to update this.

Comment (anrossi) [reply scripts/package-distribution.ps1:22]: ah, I see that now.

**Comment by thhous-msft** [comment]
@wfurt We're doing a final look at pulling this in, along with the msquic 2.0 changes. Does everything still look ok from the .NET side? Or anything else we need to add to the package?

**Comment by wfurt** [comment]
This looks ok by quick glance @thhous-msft. It fulfills IMHO the big desire e.g. ability to have multiple (major) versions  side by side. For .NET the biggest issue are breaking changes in 2.x. The changes to load and use the .so.2 are trivial. 
cc: @ManickaP

**Review by nibanks — APPROVED** [review]

**Pull request merged**